lib/rubocop/cop/rspec/file_path.rb

```diff
@@ -5,6 +5,12 @@ module Cop
     module RSpec
       # Checks that spec file paths are consistent and well-formed.
       #
+      # This cop is deprecated.
+      # We plan to remove it in the next major version update to 3.0.
+      # The migration targets are `RSpec/SpecFilePathSuffix`
+      # and `RSpec/SpecFilePathFormat`.
+      # If you are using this cop, please plan for migration.
```

Comment:
@ydah What is the suggested way to migrate?

I've updated my config like the following for `0.24.0`.

* Removed `RSpec/FilePath`
* Disabled `RSpec/SpecFilePathFormat`
* Enabled `RSpec/SpecFilePathSuffix`

But this new change in `0.24.1` is forcing me to update my config again like the following.

* Add `RSpec/FilePath` back and disable it

Does that mean my config will be broken when `RSpec/FilePath` is removed in the next major version update to 3.0?

Reply:
I understand your frustration with some reconfiguration required for a minor/patch level version dependency updates.

May I kindly ask you to elaborate what you mean by "forcing"?

You meant 2.24.0.

> my config will be broken ... in the next major version update to 3.0?

Major updates often come with breaking changes, so most probably yes.
But worry not, we'll provide a detailed update checklist like we did with [1.x -> 2.x update](https://docs.rubocop.org/rubocop-rspec/upgrade_to_version_2.html).

Reply:
Thanks for responding to my question 🙏 

> May I kindly ask you to elaborate what you mean by "forcing"?
> You meant 2.24.0.

* Before `2.24.0`: I had `RSpec/FilePath` explicitly disabled in my config.
* At `2.24.0`: I had to remove `RSpec/FilePath` completely from my config or the rubocop command would not exit with a successful status.
* At `2.24.1`: Now I had to disable `RSpec/FilePath` again or it would be enabled by default.

After updated the config twice in each of these two versions, I was a bit alerted by the words **"plan for migration"** and curious about what that actually means.

But on second thought, I think `2.24.1` just moved the state back to what we were at before `2.24.0`.

If I interpret what you said correctly, there will be no breaking changes like this until 3.0. But we probably will need to update the config when upgrading to 3.x, which is quite normal. Is my understanding correct?

Reply:
I see, thanks for the explanation. And please accept my apologies for introducing breaking changes.

Your understanding is correct.